### PR TITLE
Add client side router and link component

### PR DESCRIPTION
**Pull request by zupajul42**
Added a client side router and a link component.

Updated demo to show new features.

### CODE REVIEW

File: main.js [hunk 1]
```diff
@@ -24,35 +53,38 @@ let todoComponent = div(
...
+        () => list(todos.map((todo) => div(link(todo, "/todo/" + todo + "/info?param1=test&hello=world")))),
         () => [todos]
     )
 );
 
-let rootDiv = div(
-    h1("Hoi!"),
-    h2("Hoi2!"),
+// setup router
+useRouter().on("/", mainPage);
```
Comment (polzerj) [inline main.js:62]: Maybe also match this with "/index.html"

File: main.js [hunk 2]
```diff
@@ -24,35 +53,38 @@ let todoComponent = div(
...
-    span("Red text here :)").color("red"),
-
-    p("This is a paragraph.").on("dblclick", () => alert("Hello Universe")),
-
-    dynamic(
-        () => p(`Counter: ${counter} `, counter % 2 === 0 && "is Even"),
-        () => [counter]
-    ),
+// access router
+router()
```
Comment (polzerj) [inline main.js:65]: Is it also possible to register all routes with useRouter? What's the difference there?

File: main.js [hunk 3]
```diff
@@ -24,35 +53,38 @@ let todoComponent = div(
...
-    button("Reset").on("click", () => window.confirm("You sure?") && (counter = 0)),
-    todoComponent
-);
+        e.append(link("<- back", "/todo/"));
 
-document.body.append(rootDiv.HTML);
+        return e;
+    })
+    // redirect to active page
+    .sync();
```
Comment (polzerj) [inline main.js:90]: What does the `sync` method do? is it really necessary?

**Review by polzerj — CHANGES_REQUESTED** [review]

File: main.js (hunk 3 above)
Comment (zupajul42) [reply main.js:90]: `sync` is short for `redirect(window.location.pathname, true)`

File: main.js (hunk 2 above)
Comment (zupajul42) [reply main.js:65]: It's the same. `useRouter` returns the reference of the `Router` object. 

The only difference is that you can enable and disable the router with `useRouter(true|false)`. When the router is disabled `router()` returns null.

**Review by polzerj — APPROVED** [review]

**Review by l1ino — APPROVED** [review]

**Pull request merged**